web/react/utils/utils.jsx

```diff
@@ -420,12 +491,32 @@ module.exports.textToJsx = function(text, options) {
         var highlightSearchClass = "";
         for (var z = 0; z < words.length; z++) {
             var word = words[z];
-            var trimWord = word.replace(puncStartRegex, '').replace(puncEndRegex, '').trim();
+            var trimWord;
+            if (useMarkdown) {
+                trimWord = word.replace(endTagRegex, '').replace(startTagRegex, '').replace(puncStartRegex, '').replace(puncEndRegex, '').trim();
+            } else {
+                trimWord = word.replace(puncStartRegex, '').replace(puncEndRegex, '').trim();
+            }
             var mentionRegex = /^(?:@)([a-z0-9_]+)$/gi; // looks loop invariant but a weird JS bug needs it to be redefined here
             var explicitMention = mentionRegex.exec(trimWord);
+            console.log(word);
+            var prefix;
+            var suffix;
+            var prefixSpan;
+            var suffixSpan;
+            if (useMarkdown) {
+                prefix = (word.match(startTagRegex) ? word.match(startTagRegex) : "") + (word.replace(startTagRegex, '').match(puncStartRegex) ? word.replace(startTagRegex, '').match(puncStartRegex) : "");
+                suffix = (word.match(endTagRegex) ? word.match(endTagRegex) : "") + (word.replace(endTagRegex, '').match(puncEndRegex) ? word.replace(endTagRegex, '').match(puncEndRegex) : "");
+                prefixSpan = prefix ? (<span key={word+i+z+"pre_span"}><span dangerouslySetInnerHTML={{__html: prefix}} /></span>) : null;
+                suffixSpan = suffix ? (<span key={word+i+z+"suf_span"}><span dangerouslySetInnerHTML={{__html: suffix}} /> </span>) : (<span key={word+i+z+"suf_span"}> </span>);
```

Comment:
I didn't think about it like that, but it does in fact work!  Basically what it does is rips out the html tags from the start and/or the end of whatever the current word is and puts them in their own separate span.  Why?  Well whenever you try to do things like **@alice** or **##testing** things tended to start breaking due to how we handle making them links and such.  The easiest way I found to consistently make it work was to just remove the markdown tags from the equation while parsing these specific cases, using the prefix/suffix outline that we already had going.  The long replace/match string that also uses multiple ternaries is pretty hard to parse I admit, I need to refactor it to use if else statements for the style guide anyway so hopefully that will make it a bit clearer.  If it's still too confusing at that point I'll try to make it clearer somehow.